include/modules/hyprland/workspaces.hpp

```diff
@@ -43,19 +46,27 @@ class Workspaces : public AModule, public EventHandler {
   void update() override;
   void init();
 
+  auto all_outputs() const -> bool { return all_outputs_; }
+  auto show_special() const -> bool { return show_special_; }
+
+  auto get_bar_output() const -> std::string { return bar_.output->name; }
+
  private:
   void onEvent(const std::string&) override;
   void sort_workspaces();
-  void create_workspace(int id);
-  void remove_workspace(int id);
+  void create_workspace(const Json::Value& value);
+  void remove_workspace(std::string name);
+
+  bool all_outputs_ = false;
+  bool show_special_ = false;
 
   std::string format_;
   std::map<std::string, std::string> icons_map_;
   bool with_icon_;
-  int active_workspace_id;
+  std::string active_workspace_name;
```

Comment:
What's the reason for this change and for workspaces to remove?
Comparing integers is faster than strings.

Reply:
what you were getting from ipc was not an int it was a string, which for the normal workspaces is "id", but for named or special it won't work

Reply:
trying to use the use the name and get the id to the work with it would probably be worse

Reply:
I see.